synfig-studio/src/gui/mainwindow.cpp

```diff
@@ -98,26 +99,74 @@ MainWindow::MainWindow() :
 	bin_->add(*main_dock_book_);
 	bin_->show();
 
-	auto visible_vbox = manage(new Gtk::Box(Gtk::ORIENTATION_VERTICAL));
-	auto hidden_box   = manage(new Gtk::Box());
-
-	auto visible_menubar = App::ui_manager()->get_widget("/menubar-main");
-	auto hidden_menubar  = App::ui_manager()->get_widget("/menubar-hidden");
-	if (visible_menubar != NULL)
+	Glib::RefPtr<Gio::SimpleActionGroup> simple_action_group = Gio::SimpleActionGroup::create();
+	simple_action_group->add_action("menu-file");
+	simple_action_group->add_action("quit");
+
```

Comment:
Kindly start looking from this part ! It is just showing an empty screen without any menus.

Reply:
@rodolforg 

1. I corrected the try catch 
2. I am currently working only with quit action
3. Everything is at one place for now ( will later implemented the logic in multiple files )
4. builder is reading from string. I am aware of fact that I am not using any callback ! But will definitely implement those once a basic layout is implemented using builder. nd I am adding those actions to window using ``` insert_action_group("actions",simple_action_group);``` Is it right ?
5. segmentation fault is resolved for now ! ( null pointer exception )

Reply:
`Window` is a `Bin` container, that can only have one widget as direct child.
You are adding a (virtually) empty Box as its child; and that should be reason Window showing empty.

Gtk::Application::set_menubar() may be the way to do it in a easier way than yours.

Reply:
``` set_menubar``` only works with ``` Gtk::ApplicationWindow``` but normally we use ``` set_app_menu``` ! I tried using that but still no change ui is just the same ! File menu is still missing ! Old ui restored as expected coz I uncommented the commented normal ui code ! 

Whatever I am doing was working perfectly in my own apps !

doing everything according to [this](https://developer-old.gnome.org/gtkmm-tutorial/3.18/sec-menus-examples.html.en)

Reply:
> set_menubar only works with Gtk::ApplicationWindow

I think there should be no problem switching from `Gtk::Window` to `Gtk::ApplicationWindow`. Can you try this?

Reply:
@ice0 No issues ! I can definitely try this right away ! but still I am curious what was wrong with my code and why the g menu is not showing with the traditional method ! It should show since it was showing in my normal apps ! I am unable to find any bug in that code from past two days ! 

Feel free to tell me what was wrong with my code ? :) 

Thanks :)

Reply:
@DhairyaBahl sorry for a delay, I will take a look.

Reply:
@ice0 Ping. Kindly tell me what is wrong with my PR so I can move forward. Thanks :)

Reply:
@DhairyaBahl 
There is no menu, because you doing things in wrong order.
Please check this example: 
https://python-gtk-3-tutorial.readthedocs.io/en/latest/application.html

I think that after you test this example, it should be more clear for you what's wrong with your code. 

Also I recommend you to make a test application and post its code here so that I can show you which lines can be reordered in order to break the menu.

Reply:
@ice0 sorry for delayed response ! 

Kindly checkout this repository https://github.com/DhairyaBahl/Minus . I store all of my codes over here that I make for practice.

Secondly, Kindly look into this folder https://github.com/DhairyaBahl/Minus/tree/master/quitOrWhat ! I've stored the practice Gtk App that I've created ! 

Its was working pretty well. Kindly checkout the code. Kindly ignore the alignment issues that it may have coz I was trying and experimenting few things with the same.

Thanks :)